src/nscreg.Server.Common/Services/DataSources/UpsertUnitBulkBuffer.cs

```diff
@@ -178,7 +195,6 @@ public async Task FlushAsync()
                     var activitiesHistory = statUnitHistories.SelectMany(x => x.Activities.Distinct(new IdComparer<ActivityHistory>()))
                             .ToList();
 
-                    await _context.BulkInsertAsync(activitiesHistory, historyBulkConfig);
```

Comment:
This line should become a `AddRangeAsync`, instead of removal.

Reply:
It seems the `DataSet` is missing, you should add it.
The BulkInsert can insert even if the `DataSet` is missing.